src/console/discovery.ts

```diff
@@ -0,0 +1,151 @@
+import dgram from "dgram";
+import _ from "lodash";
+import moment from "moment";
+
+import { discoverConsoleFound, discoverConsoleLost } from "./ipc";
+
+export interface DiscoveredConsoleInfo {
+  ip: string;
+  mac: string;
+  name: string | undefined;
+  timeout: NodeJS.Timeout;
+  firstFound: moment.Moment;
+}
+
+export class ConnectionScanner {
+  public availableConnectionsByIp: { [ip: string]: DiscoveredConsoleInfo };
+  private server: dgram.Socket | null;
+
+  public constructor() {
+    this.availableConnectionsByIp = {};
+    this.server = null;
+  }
+
+  public getAvailable() {
+    return this.availableConnectionsByIp;
+  }
+
+  public getIsScanning() {
+    return !!this.server;
+  }
+
+  private _handleMessageReceive = (msg: Buffer, rinfo: dgram.RemoteInfo) => {
+    if (msg.slice(0, 10).toString() !== "SLIP_READY") {
+      // This is not a Slippi broadcast message, do nothing
+      return;
+    }
+
+    /* The structure of broadcast messages from the Wii should be:
+     *  unsigned char cmd[10]; // 0 - 10
+     *  u8 mac_addr[6]; // 10 - 16
+     *  unsigned char nickname[32]; // 16 - 48
+     */
+
+    // String for Wii's MAC address
+    const mac = [];
+    for (let i = 10; i <= 15; i += 1) {
+      mac.push(msg.readUInt8(i).toString(16).padStart(2, "0"));
+    }
+    const macAddr = `${mac[0]}:${mac[1]}:${mac[2]}:${mac[3]}:${mac[4]}:${mac[5]}`;
```

Comment:
```suggestion
    const macAddr = mac.join(":");
```
Why not just do something like this?